### PR TITLE
Custom Rules

### PR DESCRIPTION
**Pull request by erdirowlands**
WHAT

Implements custom rule logic. 

CAVEATS

Can’t use string when using custom attributes. Must use either bitstring or atom. If using a list for custom attributes, each element must be a bitstring or atom.  Rest assured if you have an `in` condition of [2,3] and you provide your attributes array of [<<“2>>, <<“3”>>] or  [<<“9>>, <<“3”>>] that will match. It was, I think, impossible to distinguish between a "string" and a "list" and because custom attributes can be both, I made the decision to not accept lists/strings. The SDK will log an error if the user provides either a string custom attribute, or if they provide a list containing string attributes (a list containing atom/bitstrings is fine). 

TESTING

- Unit tests - have included tests for the new custom rules functions, and updated the top level "Variations" tests with all rule types (in, equal, equal sensitive etc.) 
- Testgrid - all tests passing, and importantly custom rules. Pre-reqs failing as expected as not implemented yet

Screenshot 2022-11-15 at 14 24 45

RELEASE CONSIDERATIONS

Update code samples on README with how to provide custom attributes, with a note on the caveat.

### CODE REVIEW

File: rebar3.crashdump [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Error: function_clause
```
Comment (bmjen) [inline rebar3.crashdump]: This shouldn't be commited.

File: src/cfclient_evaluator.erl [hunk 2]
```diff
@@ -154,45 +157,156 @@ search_rules_for_inclusion([], _) -> not_found.
...
   case search_group_rules(TargetIdentifier, maps:get(excluded, Group, [])) of
     true ->
       excluded;
     false ->
-      search_group(included, TargetIdentifier, Group)
+      search_group(included, Target, Group)
+  end;
+search_group(included, Target, Group) ->
+  TargetIdentifier = maps:get(identifier, Target),
+  case  search_group_rules(TargetIdentifier, maps:get(included, Group, [])) of
```
Comment (bmjen) [inline src/cfclient_evaluator.erl]: Nit pick, but extra space here #OCD.

**Review by bmjen — DISMISSED** [review]

File: src/cfclient_evaluator.erl (hunk 2 above)
Comment (erdirowlands) [reply src/cfclient_evaluator.erl]: Ran the formatter, tidied this up and a few other lines.

**Comment by erdirowlands** [comment]
@bmjen  I updated the readme with code sample for Targeting with custom attributes. Would you mind taking a look to make sure it makes sense.

**Review by bmjen — APPROVED** [review]

**Pull request merged**